### ISSUE TITLE
CHD compression and folder Path for MSU-1 files

### ISSUE THREAD

**Issue by bajolzas** [bsnes-emu/bsnes#24]
Hi, is this still being developed? 
Anyways, I'd like to ask if it is possible to add CHD compression support for MSU-1 files. It would allow us to greatly reduce its size while having just one file for all track (as long as it works like it does for disc based games).

I'd also like to be able to choose a folder path for these files (like what already exists for games, patches, saves, cheats, states and screenshots) so I dont have a bunch of MSU files on my games folder.

**Comment by Screwtapello** [comment]
Yes, this is still being developed, although quite slowly - sorry it's taken so long to reply!

I'm pretty sure it's not possible to add CHD compression support for MSU-1 files. As I understand it, the CHD format stores disc images, and MSU-1 files aren't a disk image of any kind, so it's not really appropriate. You might expect that since MSU-1 files are mostly data and uncompressed audio like a CD, you should be able to treat them like CDs for the purposes of CHD compression, but I don't think that would work:

  - MSU-1 packs can contain up to 4GiB of data, CDs are limited to ~0.7GiB
  - MSU-1 packs can contain... a lot of audio tracks, CDs are limited to 99
  - MSU-1 audio tracks store a "loop point" with sample precision; CDs cannot store a loop point, and even if they did it would probably only be precise to the nearest 1/75th of a second

As a general rule, MSU-1 packs have been as succesful as they are because they work on every emulator and on the SD2SNES, and *that* is because they use uncompressed audio. Storing audio encoded as MP3s or FLACs or AAC or whatever would save a lot of disk space, but it would also be quite likely that a particular pack wouldn't work on a different system, or a different emulator, or without paying a patent licensing fee. Now that MSU-1 *is* popular it's probably less important to take such a hard-line stance, but I'm reluctant to change something that's been working so well.

On the other hand, choosing a folder path for MSU-1 files seems reasonable. You'd wind up with all the MSU-1 files for all your games all lumped together, but that's the same situation as for patches, saves, cheats, etc.

I would accept a PR for an MSU-1 path setting that was about as complex as all the other custom path settings.
